src/System.Private.Xml/src/System/Xml/Extensions/ExtensionMethods.cs

```diff
@@ -40,16 +40,12 @@ private static MethodBase FilterMethodBases(MethodBase[] methodBases, Type[] par
 
         internal static ConstructorInfo GetConstructor(this Type type, BindingFlags bindingFlags, Type[] parameterTypes)
         {
-            var constructorInfos = type.GetConstructors(bindingFlags);
-            var constructorInfo = FilterMethodBases(constructorInfos.Cast<MethodBase>().ToArray(), parameterTypes, ".ctor");
-            return constructorInfo != null ? (ConstructorInfo)constructorInfo : null;
+            return type.GetConstructor(bindingFlags, null, parameterTypes, null);
         }
 
         internal static MethodInfo GetMethod(this Type type, string methodName, BindingFlags bindingFlags, Type[] parameterTypes)
         {
-            var methodInfos = type.GetMethods(bindingFlags);
-            var methodInfo = FilterMethodBases(methodInfos.Cast<MethodBase>().ToArray(), parameterTypes, methodName);
-            return methodInfo != null ? (MethodInfo)methodInfo : null;
+            return type.GetMethod(methodName, bindingFlags, null, parameterTypes, null);
```

Comment:
There were many usages of this method unrelated to any areas I made changes. Do you want me to go and clean them up? I was trying to avoid making the diff even bigger.

---
In reply to: [122582016](https://github.com/dotnet/corefx/pull/21155#discussion_r122582016) [](ancestors = 122582016)